pkg/controller/cluster/pod.go

```diff
@@ -84,8 +84,10 @@ func (p *PodReconciler) Reconcile(ctx context.Context, req reconcile.Request) (r
 	if err := p.Client.List(ctx, &podList, listOpts); err != nil {
 		return reconcile.Result{}, ctrlruntimeclient.IgnoreNotFound(err)
 	}
+	if len(podList.Items) == 1 {
+		return reconcile.Result{}, nil
+	}
```

Comment:
Why we are now skipping the reconciliation of the single pod?

Reply:
single pod doesnt need reconciliation for etcd, because its a brand new cluster, this will probably will need some changes tho in dynamic persistence mode, but that will be another pr

Reply:
Hm, but we don't have any restriction about the scaling of the server pods. So you could eventually start from 1, then scale up to 3, and then scale down to 1 again. Maybe this particular case will work, but what if I have 3 pods, kill the `server-0` pod, that will be rescheduled with a new ip (I suppose), and then scale down to 1 pod?

I'm not sure, just thinking about some possible scenario.

Reply:
thats why actually I am listing the pods, basically everytime any scale happen to the statefulset, this controller will be triggered, listing the pod, and if the pod is marked to be deleted from the statefulset then it will be removed first from the etcd cluster member list, its a standard operation to allow a stable etcd cluster

Reply:
however if there was only one pod, and this is marked for deletion, then the controller will do nothing because its the only member in etcd member list

Reply:
Ok, I just wanted to be sure about the allowed number of replicas. I wasn't sure if we are actually allowing a change on that part, but it makes sense in my opinion.

Let's merge it, thanks!